app/Http/Middleware/LegacyInterOpAuth.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Carbon\Carbon;
+use Closure;
+
+class LegacyInterOpAuth
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (!$request->hasHeader('X-LIO-Signature') || !presence($request->query('timestamp'))) {
+            abort(403);
+        }
+
+        $timestamp = $request->query('timestamp');
+        $signature = $request->header('X-LIO-Signature');
+        $expected = hash_hmac('sha1', $request->fullUrl(), config('osu.legacy.shared_interop_secret'));
+
+        $diff = Carbon::createFromTimestamp($timestamp)->diffInSeconds();
```

Comment:
Cuz difference is absolute? >.> (e.g a 5 degree difference can mean either +5 or -5)